pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -3339,6 +3339,32 @@ type CephBlockPoolRadosNamespaceList struct {
 	Items           []CephBlockPoolRadosNamespace `json:"items"`
 }
 
+// RadosNamespaceMirroring represents the mirroring configuration of CephBlockPoolRadosNamespaceSpec
+type RadosNamespaceMirroring struct {
+	// Enabled is a flag to enable or disable mirroring
+	// +optional
+	Enabled bool `json:"enabled"`
+	// RemoteNamespace is the name of the CephBlockPoolRadosNamespace on the secondary cluster CephBlockPool
+	// +optional
+	RemoteNamespace string `json:"remoteNamespace"`
+	// Mode is the mirroring mode; either pool or image
+	// +optional
+	Mode RadosNamespaceMirroringMode `json:"mode"`
```

Comment:
Can we/ should we use a kubebuilder enum here?

Reply:
I tried but there was a complain like 
```
2024-09-30 15:26:02.299134 E | blockpool-rados-namespace-controller: failed to reconcile "rook-ceph/namespace-a" failed to add finalizer: failed to add finalizer "cephblockpoolradosnamespace.ceph.rook.io" on "namespace-a": CephBlockPoolRadosNamespace.ceph.rook.io "namespace-a" is invalid: [spec.mirroring.mode: Unsupported value: "": supported values: "pool", "image", <nil>: Invalid value: "null": some validation rules were not checked because the object was invalid; correct the existing errors to complete validation]
```

Not sure if we wanna keep this

Reply:
Hmm... Does including `""` in the enum allow it to proceed? 

We do that here:
https://github.com/rook/rook/blob/b2c39ace252b4548253c035efdcafff23ed7dd8b/pkg/apis/ceph.rook.io/v1/types.go#L2687

Reply:
it got working, thanks